### PR TITLE
WL-3857 Register sitestats updates with JMX

### PR DESCRIPTION
**Pull request by buckett**
This allows us to see how much time we are spending doing sitestats
updates.

### CODE REVIEW

**Comment by buckett** [comment]
I did the spelling correction as I was looking at contributing this back to Sakai first.

File: sitestats-impl/src/bundle/site-stats-components.xml [hunk 1]
```diff
@@ -233,14 +232,14 @@
...
-		class="org.springframework.transaction.interceptor.TransactionProxyFactoryBean">		
+		class="org.springframework.transaction.interceptor.TransactionProxyFactoryBean">
 		<property name="transactionManager" ref="@{sitestats.db}SitestatsTransactionManager" />
 		<property name="target" ref="org.sakaiproject.sitestats.api.StatsUpdateManager"/>
 		<property name="transactionAttributes">
 			<props><prop key="*">PROPAGATION_REQUIRED</prop></props>
 		</property>
 	</bean>
-	
+
```
Comment (nicholaswilson100) [inline sitestats-impl/src/bundle/site-stats-components.xml:243]: Not sure if it matters but there are 5 changes here (4 indentation changes, 1 added line of code).

Comment (buckett) [inline sitestats-impl/src/bundle/site-stats-components.xml:243]: Good spot, normally I'd fix this but this is already in Sakai as is so it's not to much of a problem having it locally.

File: sitestats-impl/src/bundle/site-stats-components.xml [hunk 2]
```diff
@@ -345,6 +344,29 @@
...
+				<property name="mappings">
+					<props>
+						<prop key="org.sakaiproject.sitestats.api.StatsUpdateManager">org.sakaiproject.sitestats.api:name=StatsUpdateManager</prop>
+					</props>
+				</property>
+			</bean>
+		</property>
+		 -->
+	</bean>
+
```
Comment (nicholaswilson100) [inline sitestats-impl/src/bundle/site-stats-components.xml:369]: Is it useful somehow adding commented out code? Might we need it later or something?

**Comment by nicholaswilson100** [comment]
Your Sakai avatar is sooo different from your weblearn.. I can't see you in the Sakai one....

File: sitestats-impl/src/bundle/site-stats-components.xml (hunk 2 above)
Comment (buckett) [inline sitestats-impl/src/bundle/site-stats-components.xml:369]: In the future the autodetected method is nicer, just at the moment it's not working.

**Comment by nicholaswilson100** [comment]
Cool

**Pull request merged**